app/src/main/java/space/taran/arknavigator/mvp/presenter/adapter/ResourcesGridPresenter.kt

```diff
@@ -28,12 +29,28 @@ class ResourcesGridPresenter(
     @Inject
     lateinit var userPreferences: UserPreferences
 
+    @Inject
+    lateinit var previewsRepo: PreviewsRepo
+
     private var resources = listOf<ResourceId>()
+    set(value) {
+        field = value
+        resourcesMeta = value.map { index.getMeta(it) }
+    }
+    private var resourcesMeta = listOf<ResourceMeta?>()
```

Comment:
This is unnecessary duplication. `ResourceId` is part of `ResourceMeta`.
Also it never can be `null`.